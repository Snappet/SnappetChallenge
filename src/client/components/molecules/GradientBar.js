import React from 'react';
import styled from 'styled-components';
import t from 'prop-types';

const Container = styled.div`
  height: 16px;
  width: 100%;
  position: relative;
  background: linear-gradient(
    90deg, rgba(255,0,0,1) 0%, rgba(232,255,0,1) 50%, rgba(14,255,0,1) 100%
  );
`;

const Marker = styled.div`
  height: 22px;
  width: 3px;
  margin-top: -4px;
  background-color: black;
  margin-left: calc(${props => (100 / props.maximun) * props.current}% - 1.5px);

  span {
    position: absolute;
    width: 20px;
    text-align: center;
    margin-left: -8px;
    margin-top: 3px;
  }
`;

const GradientBar = ({current = 0, maximun = 100}) => (
  <Container>
    <Marker current={current} maximun={maximun}>
      <span>{current?.toFixed(0)}</span>
    </Marker>
  </Container>
);

GradientBar.propTypes = {
  current: t.number,
  maximun: t.number
};

export default GradientBar;
