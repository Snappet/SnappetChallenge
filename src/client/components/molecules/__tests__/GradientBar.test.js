import React from 'react';
import renderer from 'react-test-renderer';

import GradientBar from '../GradientBar';

describe('components', () => {
  describe('molecules', () => {
    describe('GradientBar', () => {
      test('should render GradientBar with current at 0 and maximum as 200', () => {
        const tree = renderer.create(<GradientBar current={0} maximun={200} />).toJSON();
        expect(tree).toMatchSnapshot();
      });

      test('should render GradientBar with current at 50 and maximum as 100', () => {
        const tree = renderer.create(<GradientBar current={50} maximun={200} />).toJSON();
        expect(tree).toMatchSnapshot();
      });

      test('should render GradientBar with current at 100 and maximum as 100', () => {
        const tree = renderer.create(<GradientBar current={100} maximun={200} />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
