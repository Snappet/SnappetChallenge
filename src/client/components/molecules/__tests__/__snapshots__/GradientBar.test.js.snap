// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components molecules GradientBar should render GradientBar with current at 0 and maximum as 200 1`] = `
.c0 {
  height: 16px;
  width: 100%;
  position: relative;
  background: linear-gradient( 90deg,rgba(255,0,0,1) 0%,rgba(232,255,0,1) 50%,rgba(14,255,0,1) 100% );
}

.c1 {
  height: 22px;
  width: 3px;
  margin-top: -4px;
  background-color: black;
  margin-left: calc(0% - 1.5px);
}

.c1 span {
  position: absolute;
  width: 20px;
  text-align: center;
  margin-left: -8px;
  margin-top: 3px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span>
      0
    </span>
  </div>
</div>
`;

exports[`components molecules GradientBar should render GradientBar with current at 50 and maximum as 100 1`] = `
.c0 {
  height: 16px;
  width: 100%;
  position: relative;
  background: linear-gradient( 90deg,rgba(255,0,0,1) 0%,rgba(232,255,0,1) 50%,rgba(14,255,0,1) 100% );
}

.c1 {
  height: 22px;
  width: 3px;
  margin-top: -4px;
  background-color: black;
  margin-left: calc(25% - 1.5px);
}

.c1 span {
  position: absolute;
  width: 20px;
  text-align: center;
  margin-left: -8px;
  margin-top: 3px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span>
      50
    </span>
  </div>
</div>
`;

exports[`components molecules GradientBar should render GradientBar with current at 100 and maximum as 100 1`] = `
.c0 {
  height: 16px;
  width: 100%;
  position: relative;
  background: linear-gradient( 90deg,rgba(255,0,0,1) 0%,rgba(232,255,0,1) 50%,rgba(14,255,0,1) 100% );
}

.c1 {
  height: 22px;
  width: 3px;
  margin-top: -4px;
  background-color: black;
  margin-left: calc(50% - 1.5px);
}

.c1 span {
  position: absolute;
  width: 20px;
  text-align: center;
  margin-left: -8px;
  margin-top: 3px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span>
      100
    </span>
  </div>
</div>
`;
